api/src/view_function.rs

```diff
@@ -140,8 +140,14 @@ fn view_request(
         view_function.args.clone(),
         context.node_config.api.max_gas_view_function,
     );
-    let values = output.values.map_err(|err| {
-        BasicErrorWith404::bad_request_with_code_no_info(err, AptosErrorCode::InvalidInput)
+
+    let values = output.values.map_err(|status| {
```

Comment:
Let's have a test that messages are actually propagated? Eg some Move abort

Reply:
done